eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/Names.java

```diff
@@ -0,0 +1,176 @@
+
+    /**
+     * Prefix for the name.
+     */
+    private final String prefix;
+
+    /**
+     * Ctor.
+     * @param target Directory where to serialize names.
+     * @throws IOException If any issues with IO.
+     */
+    public Names(final Path target) throws IOException {
+        this.dest = target.resolve("names");
+        this.prefix = target.toString().toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]", "x");
+        this.all = Names.checked(this.dest);
+    }
+
+    /**
+     * Assign the new name to the function.
+     * @param loc Location of the insert.
+     * @return The name.
+     */
+    public String name(final String loc) {
+        final Map<String, String> cached;
+        try {
+            cached = this.all.value();
+        } catch (final IOException exc) {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Cannot load names correctly from %s",
+                    this.dest
+                ),
+                exc
+            );
+        }
+        return cached.computeIfAbsent(
+            loc,
+            key -> String.format(
+                "%s%d",
+                this.prefix,
+                cached.size()
+            )
+        );
+    }
+
+    /**
+     * Saves the function to name dispatching table.
+     * @throws IOException If any issues with IO.
+     */
+    public void save() throws IOException {
+        Files.createDirectories(this.dest.getParent());
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ObjectOutputStream oos = new ObjectOutputStream(baos);
+        oos.writeObject(this.all.value());
+        oos.flush();
+        new Home(this.dest.getParent()).save(
+            new String(Base64.getEncoder().encode(baos.toByteArray())),
+            this.dest.getFileName()
+        );
+    }
+
+    /**
+     * Prestructor to initialize this.all.
+     * @param dest Directory where to load from.
+     * @return Names map.
+     */
+    private static IoChecked<ConcurrentHashMap<String, String>> checked(final Path dest) {
+        return new IoChecked<>(
+            new Sticky<>(
+                new Synced<>(
+                    () -> {
+                        final ConcurrentHashMap<String, String> result;
+                        if (Files.exists(dest)) {
+                            result = Names.load(dest);
+                        } else {
+                            result = new ConcurrentHashMap<>();
+                        }
+                        return result;
+                    }
+                )
+            )
+        );
+    }
+
+    /**
+     * Loads the table.
+     * @param src Path where to install from.
+     * @return The map.
+     * @throws IOException If any issues with IO.
+     */
+    private static ConcurrentHashMap<String, String> load(final Path src) throws IOException {
+        try {
+            return (ConcurrentHashMap<String, String>) (
```

Comment:
@levBagryansky Unchecked cast: `java.lang.Object` to `java.util.concurrent.ConcurrentHashMap<java.lang.String,java.lang.String>`

Reply:
@volodya-lombrozo We need it because get this object from file.

Reply:
@volodya-lombrozo we can check class `ConcurrentHashMap` but it would be redundant to check that every its item insyde has type `String`.

Reply:
@levBagryansky 
> This cast is inside try catch block

The `try-catch` block won't catch a `ClassCastException`

> we can check class ConcurrentHashMap but it would be redundant to also check that every its item inside has type String.

Ok, agree. Then, just add `@SuppressWarnings("unchecked")` above the method.

Also, please, note, that it's better to place `ObjectInputStream` to `try-with-resources` block in order to close it properly:
```java
try (ObjectInputStream map = new ObjectInputStream(
            new ByteArrayInputStream(
                Base64.getDecoder().decode(
                    new FtDefault(src.getParent()).load(
                        src.getFileName().toString(),
                        ""
                    )
                )
            )
        )) {
...
```